tests/unit/multiclass/tree/RandomForest_unittest.cc

```diff
@@ -274,8 +148,187 @@ TEST(RandomForest,classify_non_nominal_test)
+
+	SGMatrix<float64_t> test_data(1, num_test, false);
+
+	for (auto i = 0; i < num_test; ++i)
+	{
+		test_data(0, i) = i < 5 ? CMath::random(0, 4) : CMath::random(6, 10);
+	}
+
+	CDenseFeatures<float64_t>* features_test =
+	    new CDenseFeatures<float64_t>(test_data);
+
+	CRandomForest* c =
+	    new CRandomForest(features_train, labels_train, num_trees, 1);
+	SGVector<bool> ft = SGVector<bool>(1);
+	ft[0] = false;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->train(features_train);
+
+	auto result = c->apply_binary(features_test);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_EQ(-1.0, res_vector[0]);
+	EXPECT_EQ(-1.0, res_vector[1]);
+	EXPECT_EQ(-1.0, res_vector[2]);
+	EXPECT_EQ(-1.0, res_vector[3]);
+	EXPECT_EQ(-1.0, res_vector[4]);
+	EXPECT_EQ(1.0, res_vector[5]);
+	EXPECT_EQ(1.0, res_vector[6]);
+	EXPECT_EQ(1.0, res_vector[7]);
+	EXPECT_EQ(1.0, res_vector[8]);
+	EXPECT_EQ(1.0, res_vector[9]);
+
+	EXPECT_NEAR(0.0, values_vector[0], 1e-1);
+	EXPECT_NEAR(0.0, values_vector[1], 1e-1);
+	EXPECT_NEAR(0.0, values_vector[2], 1e-1);
+	EXPECT_NEAR(0.0, values_vector[3], 1e-1);
+	EXPECT_NEAR(0.0, values_vector[4], 1e-1);
+	EXPECT_NEAR(1.0, values_vector[5], 1e-1);
+	EXPECT_NEAR(1.0, values_vector[6], 1e-1);
+	EXPECT_NEAR(1.0, values_vector[7], 1e-1);
+	EXPECT_NEAR(1.0, values_vector[8], 1e-1);
+	EXPECT_NEAR(1.0, values_vector[9], 1e-1);
+
+	SG_UNREF(result);
+}
+
+TEST_F(RandomForest, score_random_labels)
+{
+	// Generates data with random labels where trees don't all agree
+	sg_rand->set_seed(42);
+
+	int32_t num_train = 1000;
+	int32_t num_test = 10;
+	int32_t num_trees = 10;
+	SGMatrix<float64_t> data_B(2, num_train, false);
+
+	for (auto i = 0; i < num_train; ++i)
+	{
+		data_B(0, i) = CMath::randn_double();
+		data_B(1, i) = CMath::randn_double();
+	}
+	CDenseFeatures<float64_t>* features_train =
+	    new CDenseFeatures<float64_t>(data_B);
+
+	SGVector<float64_t> lab(num_train);
+	for (auto i = 0; i < num_train; ++i)
+	{
+		lab[i] = CMath::random(0, 1);
+	}
+
+	CMulticlassLabels* labels_train = new CMulticlassLabels(lab);
+
+	SGMatrix<float64_t> test_data(2, num_test, false);
+
+	for (auto i = 0; i < num_test; ++i)
+	{
+		test_data(0, i) = CMath::randn_double();
+		test_data(1, i) = CMath::randn_double();
+	}
+
+	CDenseFeatures<float64_t>* features_test =
+	    new CDenseFeatures<float64_t>(test_data);
+
+	CRandomForest* c =
+	    new CRandomForest(features_train, labels_train, num_trees, 2);
+	SGVector<bool> ft = SGVector<bool>(2);
+	ft[0] = false;
+	ft[1] = false;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->train(features_train);
+
+	auto result = c->apply_binary(features_test);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_NEAR(0.5, values_vector[0], 5e-1);
```

Comment:
we said we'd leave the tests as is and and iterate over them in another PR. Please correct my if there was a misunderstanding.